.github/workflows/ci.yml

```diff
@@ -1,6 +1,14 @@
 name: CI
 
-on: [push, pull_request, workflow_dispatch]
+on:
+  pull_request:
+  push:
+    branches: [master]
+  workflow_dispatch:
```

Comment:
specifying `branches` avoids double runs on any commits pushed that will additionally run on the pull_request event

Reply:
But wouldn't this limit action runs to commits pushed to "master" branch?
I.e. if I push to dev/v6.0.0, no actions would be run?

Reply:
right!
```suggestion
on:
  pull_request:
  push:
    branches: ['master', 'dev', 'v**']
  workflow_dispatch:
```